### ISSUE TITLE
--rules unusedArguments fails when an enum's associated value is assigned to the same variable name

### ISSUE THREAD

**Issue by nicorichard** [nicklockwood/SwiftFormat#1174]
It seems that `--rules unusedArguments` will fail if you use a switch's `case let` to assign an enum's associated value in such a way that it overrides a function's argument name before it is used.

Below you can find a simplified example which can reproduce the issue.

```swift
enum Foo {
    case a
    case b(String)

    func issue(name: String) -> String { // will be modified to `issue(name _: String)`
        switch self {
            case .b(let name): return name
            case .a: return name // after swiftformat modification this line will no longer compile
        }
    }

    func noIssue(name: String) -> String {
        switch self {
            case .a: return name
            case .b(let name): return name
        }
    }
}
```

**Comment by nicklockwood** [comment]
@nicorichard fixed in 0.49.8

**Comment by nicorichard** [comment]
Thanks for the quick response and your hard work 🥇
